packages/gatsby-source-graphql/README.md

```diff
@@ -150,6 +150,153 @@ module.exports = {
 }
 ```
 
+## Custom transform schema (advanced)
+
+It's possible to pass a custom transform to the schema, via a `transformSchema` option which customizes the default transform that the plugin creates before stitching the remote schema.
+
+To make it work it's basically needed to rewrite the schema transformation function, but that allows customization.
+
+```js
+// gatsby-config.js
+const transformSchema = require('./transform-schema');
+
+module.exports = {
+  plugins: [
+    {
+      resolve: "gatsby-source-graphql",
+      options: {
+        typeName: "SWAPI",
+        fieldName: "swapi",
+        url: "https://api.graphcms.com/simple/v1/swapi",
+        transformSchema,
+    },
+  ]
+}
+```
+
+The `transformSchema` function gets an object argument with the following fields:
+
+- schema (introspected remote schema)
+- link (default link)
+- resolver (default resolver)
+- defaultTransforms (an array with the default transforms)
+- options (plugin options)
+
```

Comment:
I think we need to add a note that the return value is expected to be the final schema used for stitching.

Reply:
So, basically, I should only leave this as a reference?

```js
// transform-schema.js
const { wrapSchema } = require(`@graphql-tools/wrap`)
const { linkToExecutor } = require(`@graphql-tools/links`)

const transformSchema = ({
  schema,
  link,
  resolver,
  defaultTransforms,
  options,
}) => {
  return wrapSchema(
    {
      schema,
      executor: linkToExecutor(link),
    },
    defaultTransforms
  )
}

module.exports = transformSchema
```
I just think that would be a bit vague for anyone else to do something useful without knowing at least how the `NamespaceUnderFieldTransform` is implemented. But if it's good for you, it's good for me too.

What if I link to this issue somewhere to leave a hint?

Reply:
The problem with this feature in general (as I see it) is that it requires a deep understanding of plugin internals and `graphql-tools` anyways.

I doubt that the example will be enough for anyone to actually use it. People will probably have to dig into the plugin source code anyways. But it may be rather confusing to other people who don't use this feature.

So we need to surface that this feature exists in docs and leave hints on what other knowledge people may need to actually use it.

Adding a link to the original issue makes sense to me though. Also, let's wait for the learning team review.